main_server/local_server/server_admin/c_comdef_admin_main_console.class.php

```diff
@@ -346,7 +346,7 @@ function return_main_console_html()
                 $ret .= 'var g_Save_meeting_button_name = \''.self::js_html ( $this->my_localized_strings['comdef_server_admin_strings']['meeting_save_buttonName'] ).'\';'.(defined ( '__DEBUG_MODE__' ) ? "\n" : '');
                 $ret .= 'var g_default_meeting_weekday = '.intVal ( $this->my_localized_strings['comdef_server_admin_strings']['meeting_editor_default_weekday'] ).';'.(defined ( '__DEBUG_MODE__' ) ? "\n" : '');
                 $ret .= 'var g_default_meeting_start_time = \''.self::js_html ( $this->my_localized_strings['comdef_server_admin_strings']['meeting_editor_default_start_time'] ).'\';'.(defined ( '__DEBUG_MODE__' ) ? "\n" : '');
-                $ret .= 'var g_default_meeting_duration = \''.self::js_html ( $this->my_localized_strings['comdef_server_admin_strings']['meeting_editor_default_duration'] ).'\';'.(defined ( '__DEBUG_MODE__' ) ? "\n" : '');
+                $ret .= 'var g_default_meeting_duration = \''.self::js_html ( c_comdef_server::GetLocalStrings()['default_duration_time'] ).'\';'.(defined ( '__DEBUG_MODE__' ) ? "\n" : '');
```

Comment:
Would `$this->my_localized_strings['comdef_server_admin_strings']['default_duration_time']` have worked yere?

Reply:
no that would not of worked, because we are inside that function

Reply:
Can you explain why you didn't just fix the setting of `meeting_editor_default_duration` so that it would always contain a sane default? This seems like a bit of indirection, having a default value stored in two different places, with possibly different values.

Reply:
Seems like `$this->my_localized_strings['comdef_server_admin_strings']['default_duration_time']` should exist, and if it doesn't, it's probably a problem with this line:

https://github.com/bmlt-enabled/BMLT-Root-Server/blob/master/main_server/server/c_comdef_server.class.php#L2900

Reply:
yes so if the `default_duration_time` isn't set, it was hardcoded in c_comdef_server.class.php to be 01:00:00 however this was only accessed in rare circumstances as the default was set to `meeting_editor_default_duration` and not to `default_duration_time`. I felt instead of being hardcoded it should just pull from `meeting_editor_default_duration`

Reply:
yes it should exist, but if it doesnt theyre should be a fallback default no?

Reply:
But now you're pulling from `default_duration_time`, not `meeting_editor_default_duration`. And you're pulling in a manner that is syntactically inconsistent with the lines above and below.

Reply:
oh wait hold on i was confused by your last statement

Reply:
cool, damn i greped it earlier too. should of known that

Reply:
> In PHPStorm you can do that kind of searching with shift+ctrl+f

Um, its shift+apple+f...

Reply:
wait so should I then have no fallback if  default_duration isnt set in the autoconfig anymore, or still set it and keep it at 01:00:00 or something.

Reply:
it kind of has to be setup, the only way i see it not being set is if you remove it

Reply:
as its set in the installer wizard

Reply:
I think keeping the default at 01:00:00 is fine

Reply:
I just don't like the idea of hardcoding the fallback default,however region bias is also hardcoded to us. so guess its not that big of a deal either way.

Reply:
Yeah I mean settings have to fallback to something. Either they fallback to something hardcoded or we make the software fail when it's not set. Those are basically your options.

Reply:
I would still rather fall back to `meeting_editor_default_duration` from set lang file `server_admin_strings.inc.php` instead of hardcoded in `c_comdef_admin_main_console.class.php` whats the issue with doing this?

Reply:
im just trying to get better understanding, could this fail at some point. or is the point to not pull extra resources than needed

Reply:
can I talk to my other sponsors till i get the answer I want to here?

Reply:
no that makes sense. language file should be used for language stuff and wording not settings.

Reply:
however there are ALOT of settings in those files

Reply:
actually i shouldnt say a lot, about 6

Reply:
Yeah, I'd just move them as we have a reason to. No reason to make a risky change unless there's a reason, like fixing this particular issue with this setting.

Reply:
bam, like it never existed